image_writers_guide/guidelines.adoc

```diff
@@ -9,4 +9,85 @@
+
+In addition, you should use tags in the `FROM` instruction (for example `rhel:rhel7`) will make it clear to users exactly which version of an image your image is based on.  If you use a tag other than `latest` this ensures that your image is not subjected to breaking changes that might go into the `latest` version of an upstream image.
+
+=== Do not start multiple processes
+It is not a good idea to start multiple services (like database, sshd, …​) inside one container. Containers are lighweight and can be easily linked together for multiprocess orchestration.  This also makes updates less disruptive as each image can be updated less frequently and independently.  Having a single process also makes signal handling flows clearer as you do not need to manage routing signals to spawned processes.
+
+=== Use exec in wrapper scripts
+See http://www.projectatomic.io/docs/docker-image-author-guidance/, subsection _Always `exec` in Wrapper Scripts_
+
+=== Clean temporary files
+Every temporary files you create during the build process should be removed. This includes any files added with the ADD command too.  For example it is strongly recommended you run yum clean after performing yum install operations.  If you craft your `RUN` statement as follows, you an prevent the yum cache from ending up in an image layer:
+
+	RUN yum -y install mypackage && yum clean all -y
+
+Note: Although the current Docker build process does not allow shrink the space used by the image when we remove something in a later layer, this may change in the future. 
+
+=== Place instructions in proper order
+Docker reads the Dockerfile and runs the instructions from top to bottom. Every successfully executed instruction creates a layer which can be reused next we build this (or other image). It is very important to place on top of your Dockerfile instructions that will change rarely.
+
+This will make sure the next builds of the same image will be very fast because the cache will not be invalidated by upper layer changes.
+
+=== Use EXPOSE to mark important ports
+See http://www.projectatomic.io/docs/docker-image-author-guidance/, subsection _Always `EXPOSE` Important Ports_
+
+=== Set environment variables with ENV
+It’s a good idea to set environment variables with the `ENV` instruction. One example is to set version of your project. This will make it easy for people find the version without even looking at the Dockerfile.  Another example of an environment variable worth setting is when to advertise some path on the system that could be used by some process, like `JAVA_HOME`.
+
+=== Do not set default passwords
+It is not a good idea to set default passwords. Many people will extend the image and forget to remove or change the default password. This will lead to security issues where in production there will be a user with a well-known passwords. Passwords should be configurable via environment variable instead (See anchor:configuration[])
```

Comment:
It may be ok to include a default, so that your image works when someone pulls it, but at a minimum printing a warning for users:

WARNING: Using the default password for "foo" - please set a password via the environment variable MYFOO_PASSWORD

will help encourage users to do the right thing.  If your image needs a password to generate data in a volume (like MySQL), it's even more important to convey what users should do to use your image properly.